### PR TITLE
Assure termination of client

### LINKED ISSUE #1453: Incomplete termination of client
**Describe the bug**
Client does not terminate but window closes still shown in task manager

**To Reproduce**
Happens approx. every 10th time

**Expected behavior**
Client process closes

**Screenshots**
Responsible is the following thread
```
java.lang.Thread.State: WAITING (parking)
        at jdk.internal.misc.Unsafe.park(java.base@10.0.2/Native Method)
        - parking to wait for  <0x00000000e2e9db58> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
        at java.util.concurrent.locks.LockSupport.park(java.base@10.0.2/Unknown Source)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(java.base@10.0.2/Unknown Source)
        at java.util.concurrent.LinkedBlockingDeque.takeFirst(java.base@10.0.2/Unknown Source)
        at java.util.concurrent.LinkedBlockingDeque.take(java.base@10.0.2/Unknown Source)
        at sun.nio.fs.AbstractWatchService.take(java.base@10.0.2/Unknown Source)
        at com.faforever.client.map.MapService.lambda$startDirectoryWatcher$7(MapService.java:172)
        at com.faforever.client.map.MapService$$Lambda$618/1056276267.run(Unknown Source)
        at com.github.nocatch.NoCatch.noCatch(NoCatch.java:49)
        at com.github.nocatch.NoCatch.noCatch(NoCatch.java:34)
        at com.faforever.client.map.MapService.lambda$startDirectoryWatcher$8(MapService.java:166)
        at com.faforever.client.map.MapService$$Lambda$617/1691552520.run(Unknown Source)
        at java.lang.Thread.run(java.base@10.0.2/Unknown Source)
```

**OS**
Windows

**Wanna have the bug fixed quickly?**
Visit Issue hunt...
[Issue hunt](https://issuehunt.io/r/FAForever/downlords-faf-client)

### PR DESCRIPTION
**Pull request by 1-alex98**
Fix #1453

### CODE REVIEW

File: .idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Default ignored files
+/shelf/
```
Comment (Geosearchef) [inline .idea/.gitignore:2]: is this committed on purpose?

**Review by Geosearchef — DISMISSED** [review]

File: .idea/.gitignore (hunk 1 above)
Comment (1-alex98) [reply .idea/.gitignore:2]: yeah, it is needed

File: src/main/java/com/faforever/client/map/MapService.java [hunk 2]
```diff
@@ -183,7 +183,7 @@ private Thread startDirectoryWatcher(Path mapsDirectory) {
       WatchService watcher = mapsDirectory.getFileSystem().newWatchService();
       customMapsDirectory.register(watcher, ENTRY_DELETE);
 
-      try {
+      try (watcher) {
```
Comment (Brutus5000) [inline src/main/java/com/faforever/client/map/MapService.java]: You can declare the variable in the try-with-resources block. 
`WatchService watcher = mapsDirectory.getFileSystem().newWatchService();` goes into the try()
  `    customMapsDirectory.register(watcher, ENTRY_DELETE);` becomes part of the try-block

**Review by Brutus5000 — CHANGES_REQUESTED** [review]

File: src/main/java/com/faforever/client/map/MapService.java (hunk 2 above)
Comment (1-alex98) [reply src/main/java/com/faforever/client/map/MapService.java]: oh yeah u sure, no idea why i did it like that

**Review by Geosearchef — APPROVED** [review]

**Pull request merged**